app/models/merchant.rb

```diff
@@ -4,4 +4,9 @@ class Merchant < ApplicationRecord
   has_many :transactions, through: :invoices
 
   validates :name, presence: true
+
+  def self.get_favorite_customer(merchant)
+    Customer.find(Customer.joins(:transactions, :invoices).where("invoices.merchant_id = ? and result = ?", merchant.id, "success").select("customers.*, invoices.customer_id, transactions.result").group("invoices.customer_id").order("count_id DESC").count('id').first.first)
```

Comment:
This looks very close to me! Without running it myself I have two recommendations:

1) Take off the `.count('id')`, and instead update the raw SQL that your passing as an argument to `.order`. Think about the SQL that you would write to order something by the count of a joined tables ids and I think it'll get you there.
2) Look into `limit` (an AR method), instead of `first` (which is Rubying the returned array).